### PR TITLE
Sort missing years in "No input data available for years" message

### LINKED ISSUE #1224: Minor quality of life improvement suggestion - sort "no input available for years".
Just a very minor QoL suggestion:

When a dataset is absent, ESMValTool prints out the missing years. It will look something like this:
```
No input data available for years 2048, 2049, 2050, 2051, 2052, 2053, 2054, 2055, 2056, 2057, 2058, 2059, 2060, 2061, 2062, 2063, 2064, 2065, 2066, 2067, 2068, 2069, 2070, 2071, 2072, 2073, 2074, 2075, 2076, 2077, 2078, 2079, 2080, 2081, 2082, 2083, 2084, 2085, 20
86, 2087, 2088, 2089, 2090, 2091, 2092, 2093, 2094, 2095, 2096, 2097, 2098, 2099, 2100, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 202
2, 2023, 2024, 2025, 2026, 2027, 2028, 2029, 2030, 2031, 2032, 2033, 2034, 2035, 2036, 2037, 2038, 2039, 2040, 2041, 2042, 2043, 2044
, 2045, 2046, 2047 in files ['/badc/cmip6/data/CMIP6/CMIP/MOHC/UKESM1-0-LL/historical/r12i1p1f2/Lmon/nbp/gn/latest/nbp_Lmon_UKESM1-0-
LL_historical_r12i1p1f2_gn_195001-201412.nc']


```

Can we please sort the list of years before printing it please? It would make this 1000x more readable.

### PR DESCRIPTION
**Pull request by ledm**
## Description

Closes #1224

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [ ] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/_recipe_checks.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def data_availability(input_files, var, dirnames, filenames):
     if missing_years:
         raise RecipeError(
             "No input data available for years {} in files {}".format(
-                ", ".join(str(year) for year in missing_years), input_files))
+                ", ".join(str(year) for year in sorted(missing_years)), input_files))
```
Comment (ledm) [inline esmvalcore/_recipe_checks.py:145]:
```suggestion
                ", ".join(str(year) for year in sorted(missing_years)), 
                input_files))
```

**Comment by valeriupredoi** [comment]
@ledm please don't review your own PR, I know you want to make code changes straight in the we editor but do it via the edit code option not via review, reviewing your own PR adds confusion at review stage :beer:

File: esmvalcore/_recipe_checks.py [hunk 2]
```diff
@@ -142,7 +142,8 @@ def data_availability(input_files, var, dirnames, filenames):
     if missing_years:
         raise RecipeError(
             "No input data available for years {} in files {}".format(
-                ", ".join(str(year) for year in missing_years), input_files))
+                ", ".join(str(year) for year in sorted(missing_years)),
```
Comment (jvegasbsc) [inline esmvalcore/_recipe_checks.py:145]: To fix the codacy issues:

```suggestion
                ", ".join(str(year) for year in sorted(missing_years)),
```

**Review by jvegasbsc — CHANGES_REQUESTED** [review]

**Comment by jvegasbsc** [comment]
@ledm, can you also update it with the latest changes on main? It should fix the CircleCI error

**Comment by ledm** [comment]
> @ledm, can you also update it with the latest changes on main? It should fix the CircleCI error

Done!

**Review by jvegasbsc — APPROVED** [review]

**Pull request merged**